shamil_latypov/main.cpp

```diff
@@ -79,52 +79,37 @@ class Edge {
   EdgeId id_;
   VertexId v1_;
   VertexId v2_;
-  int color_param_;
+  Color color_;
 };
 
 class Graph {
  public:
   // Возврат значений
   int get_depth() const { return depth_map_.size() - 1; }
-  int get_vertices_cnt_in_depth(int depth) const {
-    return depth_map_[depth].size();
+  const std::vector<VertexId>& get_vertices_in_depth(int depth) const {
+    return depth_map_[depth];
   }
 
   // Добавляет ребро в graph
-  void add_edge(const VertexId& v1, const VertexId& v2) {
-    assert(has_vertex_id(v1) && "Vertex 1 doesnt exist\n");
-    assert(has_vertex_id(v2) && "Vertex 2 doesnt exist\n");
-    assert(!vertices_connected(v1, v2) && "Vertices are connected\n");
-
-    int color_param = 0;
-    // По логике нашей задачи, если вторая вершина не имеет присоединенных к ней
-    // рёбер, значит добавляемое ребро серое
-    if (get_vertex(v2).get_edge_ids().size() == 0) {
-      // vertex2.depth = vertex1.depth + 1
-      get_vertex(v2).set_depth(get_vertex(v1).get_depth() + 1);
-
-      if (depth_map_.size() <= get_vertex(v2).get_depth()) {
-        depth_map_.push_back({});
-      }
-
-      // Добавление на нужную глубину vertex2.id и удаление vertex2.id из
-      // нулевой глубины
-      depth_map_[get_vertex(v2).get_depth()].push_back(v2);
-      for (auto vertex_id_it = depth_map_[0].begin();
-           vertex_id_it != depth_map_[0].end(); vertex_id_it++) {
-        if (*vertex_id_it == v2) {
-          depth_map_[0].erase(vertex_id_it);
-          break;
-        }
-      }
-    } else {  // Иначе задаем цвет ребру
-      color_param = set_edge_color(v1, v2);
+  void add_edge(const VertexId& vertex1_id, const VertexId& vertex2_id) {
+    assert(has_vertex_id(vertex1_id) && "Vertex 1 doesnt exist\n");
+    assert(has_vertex_id(vertex2_id) && "Vertex 2 doesnt exist\n");
+    assert(!vertices_connected(vertex1_id, vertex2_id) &&
+           "Vertices are connected\n");
+
+    auto& vertex1 = get_vertex(vertex1_id);
+    auto& vertex2 = get_vertex(vertex2_id);
+    const auto& color_param = set_edge_color(vertex1, vertex2);
+
+    if (color_param == Edge::Color::Gray) {
+      update_vertex_depth(vertex1, vertex2);
     }
-    const auto& new_edge =
-        edges_.emplace_back(v1, v2, get_new_edge_id(), color_param);
-    get_vertex(v1).add_edge_id(new_edge.get_id());
-    if (v1 != v2) {  // Чтобы id зеленых рёбер не повторялись
-      get_vertex(v2).add_edge_id(new_edge.get_id());
+
+    const auto& new_edge = edges_.emplace_back(vertex1_id, vertex2_id,
+                                               get_new_edge_id(), color_param);
+    vertex1.add_edge_id(new_edge.get_id());
+    if (color_param != Edge::Color::Green) {
+      vertex2.add_edge_id(new_edge.get_id());
     }
```

Comment:
Я же тебе написал пример кода. Логично, сначала создавать новую грань, а потом обновлять вершины. А ты делаешь все задом наперед.
```suggestion
    const auto& new_edge = edges_.emplace_back(vertex1_id, vertex2_id,
                                               get_new_edge_id(), color_param);
    vertex1.add_edge_id(new_edge.get_id());
    if (color_param != Edge::Color::Green) {
      vertex2.add_edge_id(new_edge.get_id());
    }
    
    if (color_param == Edge::Color::Gray) {
      update_vertex_depth(vertex1, vertex2);
    }
```